### PR TITLE
TV Show application Capstone Project

### PR DESCRIPTION
**Pull request by ibrahim777764**
### Milestone 5: TV Show application Capstone project completed

-  Used JavaScript to make websites dynamic and build basic single page apps.
- . Used ES6 syntax.
-  Used ES6 modules.
-  Used callbacks and promises.
-  Used  webpack.
-  Applied JavaScript best practices and language style guides in code.
-  Used AAA pattern for unit tests.
-  Wrote units tests for a JavaScript app.
-  Followed Gitflow.
-  Solved simple git conflicts.
-  Send and receive data from an API.
-  Used  API documentation.
-  understood and use JSON.
-  Make JavaScript code synchronous.
-  Performed a code review for a team member.
 
     **project presentation video https://www.loom.com/share/9c7a843f48e84c62af07404251f9d8cf**

### CODE REVIEW

File: src/renderer.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                  <img class='likebutton' src='https://img.icons8.com/ios-filled/50/000000/hearts.png'>
+                  <span id='likes_${shows[i].id}'>${shows[i].likes} ${shows[i].likes <= 1 ? 'Like':'Likes'}</span>
+                </figcaption>
+                <button id='comment_${shows[i].id}' class='btn btn-primary btn-md card-comment-btn comments'>Comments</button>
+                <button id='reservation_${shows[i].id}' class='btn btn-primary btn-md card-comment-btn reservations'>Reservations</button>
+            </figure>
+        </li>`
+      }
+
+      document.querySelector('.show-count').innerHTML = `Number of Shows: ${shows.length}`;
```
Comment (FrederickMih) [inline src/renderer.js:24]: - [ ] Hi @ibrahim777764, awesome job on this project. As requested from the requirements, we are expected to still create a specific function to calculate the counter numbers on each page even if the API provides that information. That should go the same with comments and reservations. We need a function though your counters are working perfectly :smiley:

File: src/modules/commentModal.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// eslint-disable-next-line import/no-unresolved
```
Comment (FrederickMih) [inline src/modules/commentModal.js:1]: - [ ] Kindly fix those errors flagged by Linters. Linters are great tools :hammer: that enable you to catch programming bugs. Avoid disabling them, please.

File: src/modules/commentModal.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  closeEl.addEventListener('click', () => { commentModalView.style.display = 'none'; });
+
+  imgEl.src = show.image;
+  showTitleEl.textContent = show.title;
+  showSummaryEl.innerHTML = show.summary;
+
+  const response = await TVShowApp.getComments(showId);
+  commentCounter.textContent = `(${typeof response.length === 'undefined' ? 0 : response.length})`;
+
+  // eslint-disable-next-line space-before-blocks
```
Comment (FrederickMih) [inline src/modules/commentModal.js:28]: - [ ] Kindly fix all programming errors flagged by Linters. Do NOT disable it. Disabling means you are changing their rules. This is not accepted.

File: src/modules/reservationModal.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line import/no-unresolved
```
Comment (FrederickMih) [inline src/modules/reservationModal.js:1]: - [ ] Kindly check comments above :point_up:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @ibrahim777764 , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



# To Highlighs

- GitHub flow has been followed :heavy_check_mark: 
- PR has a title and good summary :heavy_check_mark: 
- Linters checks are passing :heavy_check_mark: 
- Great job for implementing additional reservation feature :+1: 


## Suggested changes

- [ ] **Video Presentation**
- The video should contain a highlight about the piece of the code the student finds interesting or is proud of making.
- The topic and purpose of the project are clear 

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @ibrahim777764     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**